docs/source/dev-guide/writing-tests.md

```diff
@@ -0,0 +1,148 @@
+to learn more.
+```
+
+## Integration tests
+
+Integration tests in `conda` test the application from a high level where each test can
+potentially cover large portions of the code.  These tests may also use the local
+file system and/or perform network calls. In the following sections, we cover
+several examples of exactly how these tests look, so you can use these for
+your own tests.
+
+### Running CLI level tests
+
+CLI level tests are the highest level integration tests you can write. This means that the code in
+the test is executed as if you were running it from the command line. For example,
+you may want to write a test to confirm that an environment is created after successfully
+running `conda create`. A test like this would look like the following:
+
+```python
+import os.path
+import json
+
+from conda.testing.helpers import run_inprocess_conda_command as run
+
+TEST_ENV_NAME_1 = "test-env-1"
+
+
+def test_creates_new_environment():
+    out, err, exit_code = run(f"conda create -n {TEST_ENV_NAME_1} -y")
+
+    assert "conda activate test" in out  # ensure activation message is present
+    assert err == ""  # no error messages
+    assert exit_code == 0  # successful exit code
+
+    # Perform a separate verification that everything works using the "conda env list" command
+    out, err, exit_code = run("conda env list --json")
+    json_out = json.loads(out)
+    env_names = {os.path.basename(path) for path in json_out.get("envs", tuple())}
+
+    assert TEST_ENV_NAME_1 in env_names
+    assert err == ""
+    assert exit_code == 0
+
+    out, err, exit_code = run(f"conda remove --all -n {TEST_ENV_NAME_1}")
+
+    assert err == ""
+    assert exit_code == 0
+```
+
+Let's break down exactly what is going on in the code snippet above:
+
+First, we import a function called `run_inprocess_conda_command` (aliased to `run` here) that allows
+us to run a command using the current running process. This ends up being much more efficient and quicker than
+running this test as a subprocess.
+
+In the test itself, we first use our `run` function to create a new environment. This function
+returns the standard out, standard error and the exit code of the command. This allows us to
+perform our inspections in order to determine whether the command successfully ran.
+
+The second part of the test again uses the `run` command to call `conda env list`. This time,
+we pass the `--json` flag, which allows capturing JSON that we can better parse and more easily
+inspect. This second part of the integration test not only ensures that our previous `conda create`
+command ran successfully, but also tests to make sure that `conda env list` runs correctly.
+
+Finally, we destroy the environment we just created and ensure the standard error and the exit
+code are what we expect them to be. It is important to remember removing anything you create
+as this will be present when other tests are run.
```

Comment:
```suggestion
code are what we expect them to be. It is important to remember to remove anything you create,
as it will be present when other tests are run.
```